dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/jsonpatch/JsonPatchManager.java

```diff
@@ -104,8 +108,45 @@ public JsonPatchManager(
             if ( property.isCollection() )
             {
                 Object data = ReflectionUtils.invokeMethod( object, property.getGetterMethod() );
-                node.set( property.getCollectionName(), jsonMapper.valueToTree( data ) );
+
+                Collection collection = (Collection) data;
+
+                if ( CollectionUtils.isEmpty( collection ) )
+                {
+                    continue;
+                }
+
+                if ( BaseIdentifiableObject.class.isAssignableFrom( property.getItemKlass() ) )
```

Comment:
This should be added to the above if so we get
```java
if ( property.isCollection() && BaseIdentifiableObject.class.isAssignableFrom( property.getItemKlass() ) )
```
and the 
```java
else
                {
                    node.set( property.getCollectionName(), jsonMapper.valueToTree( data ) );
                }
```
should then apply both if it is not a collection or if you cannot do this shallow transformation based on `BaseIdentifiableObject` - otherwise you miss non collections I think

Reply:
We already do the conversion from object to jsonNode before going into the method `handleCollectionUpdates`
```
        Schema schema = schemaService.getSchema( object.getClass() );
        JsonNode node = jsonMapper.valueToTree( object );

        // since valueToTree does not properly handle our deeply nested classes,
        // we need to make another trip to make sure all collections are
        // correctly made into json nodes.
        handleCollectionUpdates( object, schema, (ObjectNode) node );
```